File: binance4j-wallet/src/main/java/com/binance4j/wallet/dto/CoinInformation.java
package com.binance4j.wallet.dto;

import java.util.List;

import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;

/**
 * The coin infos for the authenticated account.
 * 
 * @param coin              The coin abbreviation.
 * @param name              The coin name.
 * @param depositAllEnable  The deposit permission.
 * @param isLegalMoney      if the coin is a legal money.
 * @param trading           if the coin can be traded.
 * @param withdrawAllEnable The withdraw permission.
 * @param free              The free balance.
 * @param freeze            The freezed balance.
 * @param ipoable           The Initial Public Offering balance (?)
 * @param ipoing            The Initial Public Offering freezed balance (?)
 * @param locked            The locked balance.
 * @param storage           The storage
 * @param withdrawing       The withdrawing balance.
 * @param networkList       The coin network information.
 */
@ApiModel("")
public record CoinInformation(@ApiModelProperty("") String coin, @ApiModelProperty("") String name, @ApiModelProperty("") boolean depositAllEnable,
		@ApiModelProperty("") boolean isLegalMoney, @ApiModelProperty("") boolean trading, @ApiModelProperty("") boolean withdrawAllEnable,
		@ApiModelProperty("") String free, @ApiModelProperty("") String freeze, @ApiModelProperty("") String ipoable, @ApiModelProperty("") String ipoing,
		@ApiModelProperty("") String locked, @ApiModelProperty("") String storage, @ApiModelProperty("") String withdrawing,
		@ApiModelProperty("") List<CoinNetworkInformation> networkList) {
}

File: wallet/src/main/kotlin/com/binance4j/wallet/dto/CoinInformation.kt
/*
 * MIT License
 *
 * Copyright (c) 2022 Binance4j
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.binance4j.wallet.dto

import io.swagger.annotations.ApiModel
import io.swagger.annotations.ApiModelProperty

/**
 * The coin info for the authenticated account.
 *
 * @property coin              Coin abbreviation.
 * @property name              Coin name.
 * @property depositAllEnable  Deposit permission.
 * @property isLegalMoney      Is the coin a legal money?
 * @property trading           Can the coin be traded?
 * @property withdrawAllEnable Withdraw permission.
 * @property free              Free balance.
 * @property freeze            Frozen balance.
 * @property ipoable           Initial Public Offering balance (?)
 * @property ipoing            Initial Public Offering frozen balance (?)
 * @property locked            Locked balance.
 * @property storage           Storage
 * @property withdrawing       Withdrawing balance.
 * @property networkList       Coin network information.
 */
@ApiModel("The coin infos for the authenticated account.")
data class CoinInformation(
    @ApiModelProperty("Coin abbreviation.")
    var coin: String? = null,
    @ApiModelProperty("Coin name.")
    var name: String? = null,
    @ApiModelProperty("Deposit permission.")
    var depositAllEnable: Boolean? = null,
    @ApiModelProperty("Is the coin a legal money?")
    var isLegalMoney: Boolean? = null,
    @ApiModelProperty("Can the coin be traded?")
    var trading: Boolean? = null,
    @ApiModelProperty("Withdraw permission.")
    var withdrawAllEnable: Boolean? = null,
    @ApiModelProperty("Free balance.")
    var free: String? = null,
    @ApiModelProperty("Freezed balance.")
    var freeze: String? = null,
    @ApiModelProperty("Initial Public Offering balance (?)")
    var ipoable: String? = null,
    @ApiModelProperty("Initial Public Offering freezed balance (?)")
    var ipoing: String? = null,
    @ApiModelProperty("Locked balance.")
    var locked: String? = null,
    @ApiModelProperty("Storage")
    var storage: String? = null,
    @ApiModelProperty("Withdrawing balance.")
    var withdrawing: String? = null,
    @ApiModelProperty("Coin network information.")
    var networkList: List<CoinNetworkInformation>? = null
)
